// stylelint-semantic-groups/dist/es2015/properties-order.js
import { animations, border, box, clip, elementProperties, misc, scss, style, typography, visuals } from './groups';
var createGroup = function (groupName, properties) { return ({
    emptyLineBefore: 'always',
    noEmptyLineBetween: true,
    groupName: groupName,
    properties: properties,
}); };
var ruleIsArray = function (rules) { return Array.isArray(rules[0]); };
var mergeGroups = function (groups) {
    var groupRules = [];
    var _loop_1 = function (groupName, rules) {
        if (ruleIsArray(rules)) {
            groupRules.push.apply(groupRules, rules.map(function (props) { return createGroup(groupName, props); }));
        }
        else {
            groupRules.push(createGroup(groupName, rules));
        }
    };
    // eslint-disable-next-line no-restricted-syntax
    for (var _i = 0, _a = Object.entries(groups); _i < _a.length; _i++) {
        var _b = _a[_i], groupName = _b[0], rules = _b[1];
        _loop_1(groupName, rules);
    }
    return groupRules;
};
export var propertyOrdering = [
    mergeGroups({
        scss: scss,
        elementProperties: elementProperties,
        box: box,
        typography: typography,
        border: border,
        visuals: visuals,
        style: style,
        clip: clip,
        misc: misc,
        animations: animations,
    }),
    // {"unspecified": "bottomAlphabetical"}
];
