// stylelint-semantic-groups/dist/es5/groups/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.visuals = exports.typography = exports.style = exports.scss = exports.misc = exports.elementProperties = exports.clip = exports.box = exports.border = exports.animations = void 0;
var tslib_1 = require("tslib");
var animations_1 = tslib_1.__importDefault(require("./animations"));
exports.animations = animations_1.default;
var border_1 = tslib_1.__importDefault(require("./border"));
exports.border = border_1.default;
var box_1 = tslib_1.__importDefault(require("./box"));
exports.box = box_1.default;
var clip_1 = tslib_1.__importDefault(require("./clip"));
exports.clip = clip_1.default;
var element_properties_1 = tslib_1.__importDefault(require("./element-properties"));
exports.elementProperties = element_properties_1.default;
var misc_1 = tslib_1.__importDefault(require("./misc"));
exports.misc = misc_1.default;
var scss_1 = tslib_1.__importDefault(require("./scss"));
exports.scss = scss_1.default;
var style_1 = tslib_1.__importDefault(require("./style"));
exports.style = style_1.default;
var typography_1 = tslib_1.__importDefault(require("./typography"));
exports.typography = typography_1.default;
var visuals_1 = tslib_1.__importDefault(require("./visuals"));
exports.visuals = visuals_1.default;
