// stylelint-semantic-groups/dist/es5/properties-order.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.propertyOrdering = void 0;
var groups_1 = require("./groups");
var createGroup = function (groupName, properties) { return ({
    emptyLineBefore: 'always',
    noEmptyLineBetween: true,
    groupName: groupName,
    properties: properties,
}); };
var ruleIsArray = function (rules) { return Array.isArray(rules[0]); };
var mergeGroups = function (groups) {
    var groupRules = [];
    var _loop_1 = function (groupName, rules) {
        if (ruleIsArray(rules)) {
            groupRules.push.apply(groupRules, rules.map(function (props) { return createGroup(groupName, props); }));
        }
        else {
            groupRules.push(createGroup(groupName, rules));
        }
    };
    // eslint-disable-next-line no-restricted-syntax
    for (var _i = 0, _a = Object.entries(groups); _i < _a.length; _i++) {
        var _b = _a[_i], groupName = _b[0], rules = _b[1];
        _loop_1(groupName, rules);
    }
    return groupRules;
};
exports.propertyOrdering = [
    mergeGroups({
        scss: groups_1.scss,
        elementProperties: groups_1.elementProperties,
        box: groups_1.box,
        typography: groups_1.typography,
        border: groups_1.border,
        visuals: groups_1.visuals,
        style: groups_1.style,
        clip: groups_1.clip,
        misc: groups_1.misc,
        animations: groups_1.animations,
    }),
    // {"unspecified": "bottomAlphabetical"}
];
